src/Microsoft.VisualStudio.Threading/Strings.resx

```diff
@@ -156,6 +156,12 @@
   <data name="QueueEmpty" xml:space="preserve">
     <value>The queue is empty.</value>
   </data>
+  <data name="SemaphoreMisused" xml:space="preserve">
+    <value>This semaphore has been misused and can no longer be used.</value>
```

Comment:
Is there any way to indicate how the semaphore has been misused?

Reply:
A message that tells exactly what someone did wrong is used for exceptions thrown to the guilty parties.
This message is used for innocent top-level calls, who can't fix the problem.